hs_core/hydro_realtime_signal_processor.py

```diff
@@ -5,10 +5,50 @@
 import types
 from haystack.query import SearchQuerySet
 from haystack.utils import get_identifier
+from hs_core.models import SOLRQueue
 
 logger = logging.getLogger(__name__)
 
 
+def solr_update(instance): 
+    """ Update a resource's SOLR record """
+    from hs_core.models import BaseResource
+    from hs_core.search_indexes import BaseResourceIndex
+    logger.info('updating {}'.format(instance.short_id))
+    index = BaseResourceIndex()
+    newbase = BaseResource.objects.get(pk=instance.pk)
+    try:
+        index.update_object(newbase)
+    except NotHandled:
+        logger.exception("Failure: changes to %s with short_id %s not added to Solr Index.",
+                         str(type(instance)), newbase.short_id)
+
+
+def solr_delete(instance): 
+    """ Delete a resource from SOLR before deleting from Django """
+    from hs_core.models import BaseResource
+    from hs_core.search_indexes import BaseResourceIndex
+    logger.info('deleting {}'.format(instance.short_id))
+    index = BaseResourceIndex()
+    newbase = BaseResource.objects.get(pk=instance.pk)
```

Comment:
This will throw an error if the resource does not exist.  Answering my previous question about what happens in the SOLRQueue if the resource was deleted.